userspace/libscap/scap_procs.c

```diff
@@ -258,21 +258,18 @@ int32_t scap_proc_fill_info_from_stats(scap_t *handle, char* procdirname, struct
 		return SCAP_FAILURE;
 	}
 
-	if(fgets(line, sizeof(line), f) == NULL)
+	uint32_t slpos = 0;
+	while(fgets(line + slpos, sizeof(line) - slpos, f) != NULL)
 	{
-		ASSERT(false);
-		fclose(f);
-		snprintf(handle->m_lasterr, SCAP_LASTERR_SIZE, "Could not read from stat file %s (%s)",
-			 filename, scap_strerror(handle, errno));
-		return SCAP_FAILURE;
+		slpos = strlen(line);
```

Comment:
wouldn't this need to be `+=`?

Reply:
My first implementation was based on fread, but I was seeing garbage at the end of the string, like if the string was not NULL terminated, so I went with a gets loop. I Don't need the += because I just always want to position myself at the end of the current line. I did that to avoid further buffer allocation and just use line as before. I tested this with different string combinations and it seemed to work with all of them.

comm *is* limitet in size, actually pretty tightly. On my ubuntu 18, for example, it's limited to 16 characters

Reply:
Yeah, you'd need
```
if(nread == sizeof(buf))
{
    // overflow, or one byte from overflow
}
else
{
    buf[nread] = 0;
}
```

About `slpos`, I must be missing something:
1. read first line to `line+0`, 20 bytes
2. slpos = 20
3. read second line to `line+20`, 10 bytes
4. slpos = 10
5. read third line to `line+10`, 50 bytes

`line` is now the first 10 bytes of line 1, followed by line 3, which is pretty meaningless, no? Does _not_ reading into `line` (without any offset) give us anything? We only care about the last line anyway.